packages/email-sender/test/emailSenderProcessor.test.ts

```diff
@@ -0,0 +1,131 @@
+import { describe, beforeAll, vi, afterEach, it, expect } from "vitest";
+import { EmailManagerSES } from "pagopa-interop-commons";
+import {
+  AccountSuspendedException,
+  BadRequestException,
+  LimitExceededException,
+  MailFromDomainNotVerifiedException,
+  MessageRejected,
+  NotFoundException,
+  SendingPausedException,
+  TooManyRequestsException,
+} from "@aws-sdk/client-sesv2";
+import { emailSenderProcessorBuilder } from "../src/services/emailSenderProcessor.js";
+import {
+  correctEventPayload,
+  kafkaMessagePayload,
+  kafkaMessagePayloadWithValue,
+} from "./utils.js";
+
+describe("emailSenderProcessor", () => {
+  // eslint-disable-next-line functional/no-let
+  let emailSenderProcessor: ReturnType<typeof emailSenderProcessorBuilder>;
+  const mockSESEmailManager = {
+    kind: "SES",
+    send: vi.fn().mockResolvedValue(undefined),
+  } as EmailManagerSES;
+  const mockSESSender = {
+    label: "mock sender",
+    mail: "sender@mock.com",
+  };
+
+  beforeAll(async () => {
+    emailSenderProcessor = emailSenderProcessorBuilder(
+      mockSESSender,
+      mockSESEmailManager
+    );
+  });
+
+  afterEach(async () => {
+    vi.clearAllMocks();
+  });
+
+  it("should send email when message has subject, address and body", async () => {
+    const message = kafkaMessagePayload;
+    await emailSenderProcessor.processMessage(message);
+    expect(mockSESEmailManager.send).toHaveBeenCalledOnce();
+  });
+
+  it.each([
+    {
+      eventPayload: null,
+    },
+    {
+      eventPayload: { ...correctEventPayload, subject: undefined },
+    },
+    {
+      eventPayload: { ...correctEventPayload, address: undefined },
+    },
+    {
+      eventPayload: { ...correctEventPayload, body: undefined },
+    },
+    {
+      eventPayload: { ...correctEventPayload, correlationId: undefined },
+    },
+    {
+      eventPayload: { ...correctEventPayload, address: "invalid" },
+    },
+    {
+      eventPayload: { ...correctEventPayload, correlationId: "invalid" },
+    },
+  ])(
+    "should skip and log error if message is malformed: %s",
```

Comment:
here and also lines 80 and 93, since the tests don't check the logs, I think it's best to avoid mentioning the logging in the test description

alternatively, we could also test that the log is emitted, but that's a bit more complicated and maybe not worth it